core-primitives/stf-executor/src/traits.rs

```diff
@@ -31,28 +32,15 @@ pub enum StatePostProcessing {
 	Prune,
 }
 
-/// Execute shield funds on the STF
-pub trait StfExecuteShieldFunds {
-	fn execute_shield_funds(
-		&self,
-		account: AccountId,
-		amount: Amount,
-		shard: &ShardIdentifier,
-	) -> Result<H256>;
-}
+/// Operations as STF root.
+pub trait StfEnclaveSigning {
+	fn get_enclave_account(&self) -> AccountId;
 
-/// Execute a trusted call on the STF
-pub trait StfExecuteTrustedCall {
-	fn execute_trusted_call<PH>(
+	fn sign_call_with_self(
 		&self,
-		calls: &mut Vec<OpaqueCall>,
-		stf_call_signed: &TrustedOperation,
-		header: &PH,
+		trusted_call: &TrustedCall,
 		shard: &ShardIdentifier,
-		post_processing: StatePostProcessing,
-	) -> Result<Option<H256>>
-	where
-		PH: HeaderTrait<Hash = H256>;
+	) -> Result<TrustedCallSigned>;
```

Comment:
New trait (and corresponding implementation) to allow signing a trusted call with the enclave account.